### PR TITLE
Submit xmc-tools-follow-up m1

### PR DESCRIPTION
**Pull request by gmajor-encrypt**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] This pull request is being made by the same account as the accepted application.
- [x] I have disclosed any and all sources of reused code in the submitted repositories and have done my due diligence to meet its license requirements.
- [x] In case of acceptance, invoices must be submitted and payments will be transferred to the Polkadot AssetHub and/or fiat account provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://grants.web3.foundation/docs/Support%20Docs/milestone-deliverables-guidelines).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/2205

### CODE REVIEW

**Comment by PieWol** [comment]
Hey @gmajor-encrypt ,
thank you for submitting the delivery. I will notify you once my evaluation is completed.

**Comment by PieWol** [comment]
Hi @gmajor-encrypt ,
my [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/pull/1161) is finished with some changes requested. Looking forward to your comments on this and the changes you will make. Please notify me as soon as you have updates to share.

**Comment by gmajor-encrypt** [comment]
@PieWol Thanks for your evaluation. I will make change as soon.

**Comment by gmajor-encrypt** [comment]
@PieWol I have update the code. Please check again.

> Xcm V3 struct not as strictly typed as the [rust implementation](https://docs.rs/staging-xcm/latest/staging_xcm/v3/enum.Instruction.html)

Since xcm-tools does not involve the need to construct xcm messages. Therefore, dynamic types are used here, which are only useful when parse messages.

**Comment by PieWol** [comment]
Hey @gmajor-encrypt, 
thanks for the updates. Changes to the documentation are looking good. Have you made sure that the tests are working? After pulling your changes, building again and starting the tests I have the issue that the tests are still not passing.

Please let me know once you fixed the tests in Docker. I have pulled the changes and re ran the commands to build and test with Docker. If I am missing a step please let me know as well. I have updated my evaluation so it reflects the current issues and testing output.

Thanks :)

**Comment by gmajor-encrypt** [comment]
@PieWol Thanks for your evaluation. I think the lack of test success in docker is because there is no re-docker build. You can try it after building.

**Comment by PieWol** [comment]
@gmajor-encrypt I made sure the first time that I was building again. Please look at it yourself and fix the issues.

**Review by semuelle — APPROVED** [review]
I just merged the evaluation. Your milestone is hereby accepted. Congrats, @gmajor-encrypt.

File: deliveries/xcm-tools-follow-up_1.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Milestone Delivery :mailbox:
+
+**The delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/milestone-deliverables-guidelines.md).**
+
+* **Application Document:** [xcm-tools-follow-up](https://github.com/gmajor-encrypt/Grants-Program/blob/master/applications/xcm-tools-follow-up.md)
```
Comment (semuelle) [inline deliveries/xcm-tools-follow-up_1.md:5]:
```suggestion
* **Application Document:** [xcm-tools-follow-up](https://github.com/w3f/Grants-Program/blob/master/applications/xcm-tools-follow-up.md)
```

**Comment by RouvenP** [comment]
hi @gmajor-encrypt we sent the payment last Friday

**Pull request merged**